src/rpc/misc.cpp

```diff
@@ -755,6 +755,72 @@ UniValue echo(const JSONRPCRequest& request)
     return request.params;
 }
 
+std::set<uint256> parseProRegTxHashes(const UniValue& obj, int arrPos)
+{
+    if (!obj[arrPos].isArray()) throw std::runtime_error("error: mnconnect arg1 must be an array of proreg txes hashes");
+    const auto& array{obj[arrPos].get_array()};
+    std::set<uint256> vec_dmn_protxhash;
+    for (unsigned int i = 0; i < array.size(); i++) {
+        vec_dmn_protxhash.emplace(uint256S(array[i].get_str()));
+    }
+    return vec_dmn_protxhash;
+}
+
+// mnconnect command operation types
+const char* SINGLE_CONN = "single_conn";
+const char* QUORUM_MEMBERS_CONN = "quorum_members_conn";
+const char* IQR_MEMBERS_CONN = "iqr_members_conn";
+const char* PROBE_CONN = "probe_conn";
+
+/* What essentially does is add a pending MN connection
+ * Can be in the following forms:
+ * 1) Direct single DMN connection.
+ * 2) Quorum members connection (set of DMNs to connect).
+ * 3) Quorum relay members connections (set of DMNs to connect and relay intra-quorum messages).
+ * 4) Probe DMN connection.
+**/
+UniValue mnconnect(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() > 4) {
```

Comment:
it looks like the first two arguments are required:
```suggestion
    if (request.fHelp || request.params.size() < 2 || request.params.size() > 4) {
```